cmd/grpc/server.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"context"
+	"fmt"
+    "google.golang.org/grpc/status"
+    "google.golang.org/grpc/codes"
+    "google.golang.org/grpc/metadata" 
+	pb "github.com/PerfectStepCoder/shorturl/internal/proto/gen"
+	"github.com/PerfectStepCoder/shorturl/internal/storage"
+    "google.golang.org/protobuf/types/known/emptypb"
+)
+
+// UsersServer поддерживает все необходимые методы сервера.
+type ShorterServerGRPC struct {
+	pb.UnimplementedShorterServer
+    mainStorage storage.PersistanceStorage
+}
+
+// ShorterURL - обработчик ссылок.
+func (s *ShorterServerGRPC) ShorterURL(ctx context.Context, in *pb.RequestFullURL) (*pb.ResponseShortURL, error) {
+    var response pb.ResponseShortURL
+    var userUID string
+    
+    if md, ok := metadata.FromIncomingContext(ctx); ok {
+        values := md.Get("userUID")
+        if len(values) > 0 {
+            userUID = values[0]
+        }
+    } else {
+        return &response, status.Errorf(codes.Internal, `We didnot catch userUID`)
```

Comment:
Если есть возможность стоит проверять "негативный" кейс и возвращать управление - так можно обойтись без `if-else`.  
```suggestion
    if md, ok := metadata.FromIncomingContext(ctx); !ok {
        return &response, status.Errorf(codes.Internal, `We didnot catch userUID`)
    }
    
    values := md.Get("userUID")
    if len(values) > 0 {
        userUID = values[0]
    }
```

Reply:
Готово, применил твой код